lib/epubmaker/content.rb

```diff
@@ -34,31 +34,18 @@ def inspect
     end
 
     # :call-seq:
-    #    initialize(file, id, media, title, level, notoc)
-    #    initialize(hash)
-    # Construct Content object by passing a sequence of parameters or hash.
-    # Keys of +hash+ relate with each parameters.
-    # +file+ (or +hash+["file"]) is required. Others are optional.
-    def initialize(fileorhash, id = nil, media = nil, title = nil, level = nil, notoc = nil, properties = nil, chaptype = nil) # rubocop:disable Metrics/ParameterLists
-      if fileorhash.instance_of?(Hash)
-        @id = fileorhash['id']
-        @file = fileorhash['file']
-        @media = fileorhash['media']
-        @title = fileorhash['title']
-        @level = fileorhash['level']
-        @notoc = fileorhash['notoc']
-        @properties = fileorhash['properties'] || []
-        @chaptype = fileorhash['chaptype']
-      else
-        @file = fileorhash
-        @id = id
-        @media = media
-        @title = title
-        @level = level
-        @notoc = notoc
-        @properties = properties || []
-        @chaptype = chaptype
-      end
+    #    initialize(params)
+    # Construct Content object by passing named parameters.
+    # +params+[:file] is required. Others are optional.
+    def initialize(params)
+      @id = params[:id]
+      @file = params[:file]
+      @media = params[:media]
+      @title = params[:title]
+      @level = params[:level]
+      @notoc = params[:notoc]
+      @properties = params[:properties] || []
+      @chaptype = params[:chaptype]
```

Comment:
```suggestion
    def initialize(file:, id: nil, media: nil, title: nil, level: nil, notoc: nil, properties: nil, chaptype: nil)
      @id = id
      @file = file
      @media = media
      @title = title
      @level = level
      @notoc = notoc
      @properties = properties || []
      @chaptype = chaptype
```

こちらはこんな感じでしょうか。コメントの通り、fileがないとエラーになるはず。

Reply:
はいー